sdkproject/Assets/Mapbox/Unity/MeshGeneration/Modifiers/MeshModifiers/SnapTerrainRaycastModifier.cs

```diff
@@ -19,57 +21,31 @@ public override ModifierType Type
 
 		public override void Run(VectorFeatureUnity feature, MeshData md, UnityTile tile = null)
 		{
-			// TODO: Get this from tile as IMapReadable.
-			float worldScale = FindObjectOfType<AbstractMap>().WorldRelativeScale;
+			scaledX = tile.Rect.Size.x * tile.TileScale;
+			scaledY = tile.Rect.Size.y * tile.TileScale;
 
-			if (md.Vertices.Count > 0)
+			foreach (var sub in feature.Points)
 			{
-				for (int i = 0; i < md.Vertices.Count; i++)
+				for (int i = 0; i < sub.Count; i++)
 				{
-					var h = tile.QueryHeightData((float)((md.Vertices[i].x + tile.Rect.Size.x / 2) / tile.Rect.Size.x),
-						(float)((md.Vertices[i].z + tile.Rect.Size.y / 2) / tile.Rect.Size.y));
+					var h = tile.QueryHeightData((float)((sub[i].x + scaledX / 2) / scaledX), (float)((sub[i].z + scaledY / 2) / scaledY));
```

Comment:
@brnkhy This does not take into consideration the terrain height multiplier, does it?

Reply:
@david-rhodes it should I believe, can't see why not. multiplied height effects queried height anyway right?